### PR TITLE
workround for podman readlink error

### PR DESCRIPTION
**Pull request by luluz66**
When we detect that podman failed with the storage corruption warning
and readlink no such file error, we will remove the image and set the
command result to unavailable error to trigger a retry.

### CODE REVIEW

File: enterprise/server/remote_execution/containers/podman/podman.go [hunk 1]
```diff
@@ -164,6 +173,13 @@ func (c *podmanCommandContainer) Create(ctx context.Context, workDir string) err
 	if err = createResult.Error; err != nil {
 		return status.UnavailableErrorf("failed to create container: %s", err)
 	}
+	if isStorageCorrupted(createResult) {
```
Comment (tylerwilliams) [inline enterprise/server/remote_execution/containers/podman/podman.go:176]: does this need to happen before the check on line 173?

File: enterprise/server/remote_execution/containers/podman/podman.go [hunk 2]
```diff
@@ -141,6 +143,13 @@ func (c *podmanCommandContainer) Run(ctx context.Context, command *repb.Command,
 	podmanRunArgs = append(podmanRunArgs, c.image)
 	podmanRunArgs = append(podmanRunArgs, command.Arguments...)
 	result = runPodman(ctx, "run", nil, nil, podmanRunArgs...)
+	if isStorageCorrupted(result) {
+		result.Error = status.UnavailableError("podman run failed because of storage corruption")
+		err = removeImage(ctx, c.image)
+		if err != nil {
+			log.Warningf("Failed to remove corrupted image: %s", err)
+		}
```
[the remark this answers is not in the document]
Comment (luluz66) [reply enterprise/server/remote_execution/containers/podman/podman.go]: Done

File: enterprise/server/remote_execution/containers/podman/podman.go [hunk 3]
```diff
@@ -141,6 +143,13 @@ func (c *podmanCommandContainer) Run(ctx context.Context, command *repb.Command,
 	podmanRunArgs = append(podmanRunArgs, c.image)
 	podmanRunArgs = append(podmanRunArgs, command.Arguments...)
 	result = runPodman(ctx, "run", nil, nil, podmanRunArgs...)
+	if isStorageCorrupted(result) {
+		result.Error = status.UnavailableError("podman run failed because of storage corruption")
```
[the remark this answers is not in the document]
Comment (luluz66) [reply enterprise/server/remote_execution/containers/podman/podman.go]: Done

File: enterprise/server/remote_execution/containers/podman/podman.go [hunk 4]
```diff
@@ -287,3 +296,31 @@ func killContainerIfRunning(ctx context.Context, containerName string) error {
...
+// More details can be found at https://github.com/containers/storage/issues/1136. When this
+// happens when need to remove the image before re-pulling the image in order to fix it.
+func (c *podmanCommandContainer) maybeCleanupCorruptedImages(ctx context.Context, result *interfaces.CommandResult) error {
+	if result.ExitCode != podmanInternalExitCode {
+		return nil
+	}
+	if !storageErrorRegex.MatchString(string(result.Stderr)) {
+		return nil
+	}
+	result.Error = status.UnavailableError("a storage corruption occurred for podman")
```
Comment (bduffany) [inline enterprise/server/remote_execution/containers/podman/podman.go:310]: Set `result.ExitCode = commandutil.NoExitCode` here (the convention for CommandResult is that `ExitCode>=0` if and only if `Error==nil`)

**Review by bduffany — APPROVED** [review]
LGTM, one nit

File: enterprise/server/remote_execution/containers/podman/podman.go (hunk 4 above)
Comment (luluz66) [reply enterprise/server/remote_execution/containers/podman/podman.go:310]: Done. Thanks for the explanation!

**Review by tylerwilliams — APPROVED** [review]

**Pull request merged**